src/wp-includes/embed.php

```diff
@@ -1059,18 +1059,16 @@ function enqueue_embed_scripts() {
 }
 
 /**
- * Prints the CSS in the embed iframe header.
+ * Enqueue the CSS in the embed iframe header.
  *
- * @since 4.4.0
+ * @since 6.4.0
  */
-function print_embed_styles() {
-	$type_attr = current_theme_supports( 'html5', 'style' ) ? '' : ' type="text/css"';
-	$suffix    = SCRIPT_DEBUG ? '' : '.min';
-	?>
-	<style<?php echo $type_attr; ?>>
-		<?php echo file_get_contents( ABSPATH . WPINC . "/css/wp-embed-template$suffix.css" ); ?>
-	</style>
-	<?php
+function wp_enqueue_embed_styles() {
+	if ( ! has_action( 'embed_head', 'print_embed_styles' ) ) {
+		return;
+	}
+	remove_action( 'embed_head', 'print_embed_styles' );
+	wp_enqueue_style( 'wp-embed-template' );
```

Comment:
This is a change in behavior because now the `wp-embed-template` _may_ not be inlined. I know that, in many cases, it would still be inlined because of the logic to inline stylesheets up to a certain limit, but still there is now a chance that it is _not_ inlined. If we wanted to keep the original behavior, we could still register an "empty" stylesheet, then get the file contents and use `wp_add_inline_style()`.

I'm not saying the change in behavior here is wrong, but I want to make sure it is intentional. Why was this inlined before, and why is it now okay to not have it inlined in every situation?

Reply:
@felixarntz Good call out. To be overly safe, I have converted this to use `wp_add_inline_style`. Just in case the inliner doesnt work for some reason.